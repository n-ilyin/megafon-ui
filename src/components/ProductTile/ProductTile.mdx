---
name: ProductTile
menu: Components
---

import { Playground, PropsTable } from 'docz';
import ProductTile from './ProductTile';
import MegafonTv from 'icons/Service-logos/24/MF-TV.svg';

export const Wrapper = ({children}) => <div style={{ padding: '20px', display: 'flex', flexWrap: 'wrap' }}>{children}</div>

# ProductTile

## Properties ProductTile
<PropsTable of={ProductTile} />

## Basic usage

export const tariff = {
    className: 'custom-class',
    title: 'Нзвание тарифа',
    topBadgeTitle: 'Тариф-конструктор от 450₽',
    secondParamsHead: 'Дополнительные возможности:',
    shopTag: 'choose',
    startCallsIndex: 1,
    startTrafficIndex: 1,
    link: '#',
    moreLinkText: 'Подробнее',
    showMoreLink: true,
    buyLink: '#',
    usePackBuyLink: true,
    buyButtonText: 'купить',
    showBuyButton: true,
    buttonBorder: 'green',
    buttonFontColor: 'white',
    buttonPassiveColor: 'green',
    connectLink: '#',
    connectButtonText: 'Подключить',
    showConnectButton: true,
    payment: {
        title: 'При покупке новой SIM–карты',
        discount: '',
        unit: '₽ за 30 дней',
        unitExtra: 'за 30 дней',
        unitValue: '₽',
        value: '100',
    },
    packs: [
        {
            isDelim: false,
            title: 'Звонки на все номера России',
            unit: 'минут',
            value: 1500,
        },
        {
            isDelim: true,
            title: '"Безлимитный интернет"',
            unit: '',
            value: 0,
        }
    ],
    secondParams: [
        {
            footnote: 'Участвуйте в программе и тратьте кэшбэк на сервисы и покупки в салонах и интернет-магазине МегаФона',
            svgIcon: <MegafonTv />,
            title: 'Кэшбэк 20%',
            unit: '',
            value: '',
        },
        {
            footnote: '',
            svgIcon: <MegafonTv />,
            title: 'Безлимитные SMS',
            unit: '₽ за 30 дней',
            value: '55',
        },
    ],
    cashback: {
        title: 'Кэшбэк',
        unit: '%',
        value: 20,
    },
    firstParams: {
        title: 'Первые параметры',
        caption: '',
        items: [
            {
                caption: 'Особый статус и привилегии',
                svgIcon: <MegafonTv />,
                title: 'VIP-тариф',
            },
            {
                caption: 'Кино, ТВ и сериалы без доплаты',
                svgIcon: <MegafonTv />,
                title: 'МегаФон ТВ',
            },
        ],
    },
    info: {
        billingIds: ['123', '123'],
        fast: true,
        id: 2094,
        ivr: '05007897',
        lkLink: 'https://lk.megafon.ru/tariffs/',
        popupText: null,
        sms: '05007897',
        smsSite: null,
        smsText: 'ДА',
        title: 'Включайся! Смотри',
        ussd: '<span class=\"c-number-call\"><b class=\"asterisk\">*</b></span>',
    },
};

export const servicePacks = [
    {
        buyLink: '/zakaz/?tariff=choose_11',
        options: [
            {
                footnote: 'Участвуйте в программе и тратьте кэшбэк на сервисы',
                svgIcon: <MegafonTv />,
                title: 'Кэшбэк 1%',
                unit: '',
                value: '',
            },
            {
                footnote: '',
                svgIcon: <MegafonTv />,
                title: 'Безлимитные SMS',
                unit: '₽ за 30 дней',
                value: '1',
            }
        ],
        payment: {
            discount: '',
            value: '250',
        },
        shopTag: 'choose_11',
        calls: {
            unit: 'минут',
            value: 300
        },
        traffic: {
            unit: 'ГБ',
            value: 6,
        },
    },
    {
        buyLink: '/zakaz/?tariff=choose_11',
        options: [
            {
                footnote: 'Участвуйте в программе и тратьте кэшбэк на сервисы',
                svgIcon: <MegafonTv />,
                title: 'Кэшбэк 2%',
                unit: '',
                value: '',
            },
            {
                footnote: '',
                svgIcon: <MegafonTv />,
                title: 'Безлимитные SMS',
                unit: '₽ за 30 дней',
                value: '2',
            }
        ],
        payment: {
            discount: '',
            value: '450',
        },
        shopTag: 'choose_11',
        calls: {
            unit: 'минут',
            value: 300
        },
        traffic: {
            unit: 'ГБ',
            value: 18,
        },
    },
    {
        buyLink: '/zakaz/?tariff=choose_11',
        options: [
            {
                footnote: 'Участвуйте в программе и тратьте кэшбэк на сервисы',
                svgIcon: <MegafonTv />,
                title: 'Кэшбэк 3%',
                unit: '',
                value: '',
            },
            {
                footnote: '',
                svgIcon: <MegafonTv />,
                title: 'Безлимитные SMS',
                unit: '₽ за 30 дней',
                value: '3',
            }
        ],
        payment: {
            discount: '',
            value: '150',
        },
        shopTag: 'choose_11',
        calls: {
            unit: 'минут',
            value: 650
        },
        traffic: {
            unit: 'ГБ',
            value: 6,
        },
    },
    {
        buyLink: '/zakaz/?tariff=choose_11',
        options: [
            {
                footnote: 'Участвуйте в программе и тратьте кэшбэк на сервисы',
                svgIcon: <MegafonTv />,
                title: 'Кэшбэк 4%',
                unit: '',
                value: '',
            },
            {
                footnote: '',
                svgIcon: <MegafonTv />,
                title: 'Безлимитные SMS',
                unit: '₽ за 30 дней',
                value: '4',
            }
        ],
        payment: {
            discount: '',
            value: '350',
        },
        shopTag: 'choose_11',
        calls: {
            unit: 'минут',
            value: 650
        },
        traffic: {
            unit: 'ГБ',
            value: 18,
        },
    },
];

<Playground wrapper={Wrapper}>
    <div style={{ maxWidth: '30%', minWidth: '330px', margin: '0 20px 20px', display: 'flex' }}>
        <ProductTile {...tariff} buyButtonPassiveColor="transparent-green"/>
    </div>
    <div style={{ maxWidth: '30%', minWidth: '330px', margin: '0 20px 20px', display: 'flex' }}>
        <ProductTile {...tariff} servicePacks={servicePacks} />
    </div>
</Playground>

## Included props

``` javascript
const tariff = {
    className: 'custom-class',
    title: 'Нзвание тарифа',
    topBadgeTitle: 'Тариф-конструктор от 450₽',
    secondParamsHead: 'Дополнительные возможности:',
    shopTag: 'choose',
    startCallsIndex: 1,
    startTrafficIndex: 1,
    link: '#',
    moreLinkText: 'Подробнее',
    showMoreLink: true,
    buyLink: '#',
    usePackBuyLink: true,
    buyButtonText: 'купить',
    showBuyButton: true,
    buttonBorder: 'green',
    buttonFontColor: 'white',
    buttonPassiveColor: 'green',
    connectLink: '#',
    connectButtonText: 'Подключить',
    showConnectButton: true,
    payment: {
        title: 'При покупке новой SIM–карты',
        discount: '',
        unit: '₽ за 30 дней',
        unitExtra: 'за 30 дней',
        unitValue: '₽',
        value: '100',
    },
    packs: [
        {
            isDelim: false,
            title: 'Звонки на все номера России',
            unit: 'минут',
            value: 1500,
        },
        {
            isDelim: true,
            title: '"Безлимитный интернет"',
            unit: '',
            value: 0,
        }
    ],
    secondParams: [
        {
            footnote: 'Участвуйте в программе и тратьте кэшбэк на сервисы и покупки в салонах и интернет-магазине МегаФона',
            svgIcon: <MegafonTv />,
            title: 'Кэшбэк 20%',
            unit: '',
            value: '',
        },
        {
            footnote: '',
            svgIcon: <MegafonTv />,
            title: 'Безлимитные SMS',
            unit: '₽ за 30 дней',
            value: '55',
        },
    ],
    cashback: {
        title: 'Кэшбэк',
        unit: '%',
        value: 20,
    },
    firstParams: {
        title: 'Первые параметры',
        caption: '',
        items: [
            {
                caption: 'Особый статус и привилегии',
                svgIcon: <MegafonTv />,
                title: 'VIP-тариф',
            },
            {
                caption: 'Кино, ТВ и сериалы без доплаты',
                svgIcon: <MegafonTv />,
                title: 'МегаФон ТВ',
            },
        ],
    },
    info: {
        billingIds: ['123', '123'],
        fast: true,
        id: 2094,
        ivr: '05007897',
        lkLink: 'https://lk.megafon.ru/tariffs/',
        popupText: null,
        sms: '05007897',
        smsSite: null,
        smsText: 'ДА',
        title: 'Включайся! Смотри',
        ussd: '<span class=\"c-number-call\"><b class=\"asterisk\">*</b></span>',
    },
};

const servicePacks = [
    {
        buyLink: '/zakaz/?tariff=choose_11',
        options: [
            {
                footnote: 'Участвуйте в программе и тратьте кэшбэк на сервисы',
                svgIcon: <MegafonTv />,
                title: 'Кэшбэк 1%',
                unit: '',
                value: '',
            },
            {
                footnote: '',
                svgIcon: <MegafonTv />,
                title: 'Безлимитные SMS',
                unit: '₽ за 30 дней',
                value: '1',
            }
        ],
        payment: {
            discount: '',
            value: '250',
        },
        shopTag: 'choose_11',
        calls: {
            unit: 'минут',
            value: 300
        },
        traffic: {
            unit: 'ГБ',
            value: 6,
        },
    },
    {
        buyLink: '/zakaz/?tariff=choose_11',
        options: [
            {
                footnote: 'Участвуйте в программе и тратьте кэшбэк на сервисы',
                svgIcon: <MegafonTv />,
                title: 'Кэшбэк 2%',
                unit: '',
                value: '',
            },
            {
                footnote: '',
                svgIcon: <MegafonTv />,
                title: 'Безлимитные SMS',
                unit: '₽ за 30 дней',
                value: '2',
            }
        ],
        payment: {
            discount: '',
            value: '450',
        },
        shopTag: 'choose_11',
        calls: {
            unit: 'минут',
            value: 300
        },
        traffic: {
            unit: 'ГБ',
            value: 18,
        },
    },
    {
        buyLink: '/zakaz/?tariff=choose_11',
        options: [
            {
                footnote: 'Участвуйте в программе и тратьте кэшбэк на сервисы',
                svgIcon: <MegafonTv />,
                title: 'Кэшбэк 3%',
                unit: '',
                value: '',
            },
            {
                footnote: '',
                svgIcon: <MegafonTv />,
                title: 'Безлимитные SMS',
                unit: '₽ за 30 дней',
                value: '3',
            }
        ],
        payment: {
            discount: '',
            value: '150',
        },
        shopTag: 'choose_11',
        calls: {
            unit: 'минут',
            value: 650
        },
        traffic: {
            unit: 'ГБ',
            value: 6,
        },
    },
    {
        buyLink: '/zakaz/?tariff=choose_11',
        options: [
            {
                footnote: 'Участвуйте в программе и тратьте кэшбэк на сервисы',
                svgIcon: <MegafonTv />,
                title: 'Кэшбэк 4%',
                unit: '',
                value: '',
            },
            {
                footnote: '',
                svgIcon: <MegafonTv />,
                title: 'Безлимитные SMS',
                unit: '₽ за 30 дней',
                value: '4',
            }
        ],
        payment: {
            discount: '',
            value: '350',
        },
        shopTag: 'choose_11',
        calls: {
            unit: 'минут',
            value: 650
        },
        traffic: {
            unit: 'ГБ',
            value: 18,
        },
    },
];
```
