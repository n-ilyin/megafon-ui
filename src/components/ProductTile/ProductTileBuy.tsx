import * as React from 'react';
import { cnCreate } from '../../utils/cn';
import './style/ProductTileBuy.less';
import Button from '../Button/Button';
import TextLink from '../TextLink/TextLink';
import { IProductTileBuy } from './types';
import { productTileBuyProps } from './propTypes';

const cn = cnCreate('mfui-product-tile-buy');
class ProductTileBuy extends React.Component<IProductTileBuy> {
    static propTypes = productTileBuyProps;

    render() {
        const {
            className,
            buyButton = {},
            connectButton = {},
            onClickBuy,
            onClickConnect,
        } = this.props;
        const {
            buyLink,
            buyButtonText,
            buttonPassiveColor,
            showBuyButton,
        } = buyButton;
        const {
            connectLink,
            connectButtonText,
            showConnectButton,
        } = connectButton;

        return (
            <div className={cn('', {}, className)}>
                {showBuyButton &&
                    <Button
                        className={cn('button', { 'single-button': !showConnectButton })}
                        passiveColor={buttonPassiveColor}
                        hoverColor="green"
                        sizeAll="medium"
                        href={buyLink}
                        onClick={onClickBuy}
                    >
                        {buyButtonText}
                    </Button>
                }
                {showConnectButton &&
                    <TextLink
                        className={cn('detail-link')}
                        href={connectLink}
                        target="_blank"
                        onClick={onClickConnect}
                    >
                        {connectButtonText}
                    </TextLink>
                }
            </div>
        );
    }
}

export default ProductTileBuy;
