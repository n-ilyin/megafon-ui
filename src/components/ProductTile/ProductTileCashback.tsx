import * as React from 'react';
import { cnCreate } from '../../utils/cn';
import './style/ProductTileCashback.less';
import { ICashback } from './types';
import { productTileCashbackProps } from './propTypes';

const cn = cnCreate('mfui-product-tile-cashback');
class ProductTileCashback extends React.Component<Partial<ICashback>> {
    static propTypes = productTileCashbackProps;

    render() {
        const { title, value, unit } = this.props;

        if (!title && !value && !unit) {
            return null;
        }

        return (
            <div className={cn('')}>
                <span className={cn('text')}>
                    {title} {value} {unit}
                </span>
            </div>
        );
    }
}

export default ProductTileCashback;
