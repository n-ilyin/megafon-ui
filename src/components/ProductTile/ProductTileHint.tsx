import * as React from 'react';
import { cnCreate } from '../../utils/cn';
import './style/ProductTileHint.less';
import { IProductTileHint } from './types';
import { productTileHintProps } from './propTypes';

const cn = cnCreate('mfui-product-tile-hint');
class ProductTileHint extends React.Component<IProductTileHint> {
    static propTypes = productTileHintProps;
    render() {
        const { title } = this.props;

        return (
            <div className={cn('')}>
                <div className={cn('text')}>
                    <span className={cn('description')}>{title}</span>
                </div>
                <div className={cn('border')}/>
            </div>
        );
    }
}

export default ProductTileHint;
