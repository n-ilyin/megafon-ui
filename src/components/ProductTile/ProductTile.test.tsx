import * as React from 'react';
import { shallow, mount } from 'enzyme';
import ProductTile from './ProductTile';
import MegafonTv from 'icons/Service-logos/24/MF-TV.svg';
import Buy from './ProductTileBuy';
import Button from '../Button/Button';
import TextLink from '../TextLink/TextLink';

const info = {
    billingIds: [ '1223' ],
    fast: false,
    sms: null,
    smsText: null,
    smsSite: null,
    ivr: null,
    ussd: null,
    lkLink: 'https://lk.megafon.ru/tariffs/',
    popupText: null,
    title: 'Включайся! Выбирай',
    id: 2098,
};

const tariff = {
    title: 'Включайся! Выбирай',
    link: '/tariffs/all/vybiray.html',
    description: 'description',
    isBuyable: true,
    buyLink: '/zakaz/?tariff=choose',
    buyButtonText: 'Купить',
    buttonPassiveColor: 'green' as 'green',
    shopTag: 'choose',
    connectLink: '/tariffs/all/vybiray.html',
    showConnectButton: true,
    payment: {
        title: 'При покупке новой SIM–карты',
        value: '3000',
        unitExtra: 'за 30 дней',
        unitValue: '₽',
        unit: '₽ за 30 дней',
        discount: '100',
    },
    packs: [
        {
            value: 10,
            unit: 'ГБ',
            title: 'Звонки на все номера России',
            isDelim: false,
        },
        {
            value: 0,
            unit: '',
            title: 'Безлимитный интернет',
            isDelim: true,
        },
    ],
    firstParams: {
        title: 'Параметры',
        caption: 'Тариф для планшетов',
        items: [{
            caption: 'Пакеты кино и сериалов',
            svgIcon: <MegafonTv />,
            title: 'МегаФон ТВ',
        }],
    },
    secondParams: [
        {
            footnote: 'Участвуйте в программе',
            svgIcon: <MegafonTv />,
            title: 'Кэшбэк 20%',
            unit: '',
            value: '',
        },
        {
            footnote: '',
            svgIcon: <MegafonTv />,
            title: 'Безлимитные SMS',
            unit: '₽ за 30 дней',
            value: '100',
        },
    ],
    cashback: {
        title: 'Кэшбэк',
        value: 20,
        unit: '%',
    },
    info,
};

const servicePacks = [
    {
        buyLink: '/zakaz/?tariff=choose_11',
        shopTag: 'choose_11',
        calls: {
            value: 300,
            unit: 'минут',
        },
        traffic: {
            value: 6,
            unit: 'ГБ',
        },
        payment: {
            value: '450',
            discount: '',
        },
        options: [
            {
                title: 'Кэшбэк 20%',
                value: '',
                unit: '',
                footnote: 'Участвуйте в программе',
                svgIcon: <MegafonTv />,
            },
            {
                title: 'Безлимитные SMS',
                value: '55',
                unit: '₽ за 30 дней',
                footnote: 'Подключите в разделе «Услуги и опции»',
                svgIcon: <MegafonTv />,
            },
        ],
    },
];

const compareRestState = wrapper => {
    const [ currentPack ] = servicePacks;
    expect(wrapper.state('buyLink')).toEqual(currentPack.buyLink);
    expect(wrapper.state('options')).toEqual(currentPack.options);
    expect(wrapper.state('discount')).toEqual(currentPack.payment.discount);
    expect(wrapper.state('price')).toEqual(currentPack.payment.value);
    expect(wrapper.state('currentPack')).toEqual(currentPack);
};

describe('<ProductTile />', () => {
    it('renders ProductTile', () => {
        const wrapper = shallow(<ProductTile {...tariff} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders ProductTile', () => {
        const wrapper = shallow(<ProductTile {...tariff} servicePacks={servicePacks} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders ProductTile with topBadgeTitle', () => {
        const wrapper = shallow(
            <ProductTile
                {...tariff}
                servicePacks={servicePacks}
                topBadgeTitle="Тариф-конструктор от 450 ₽"
            />
        );

        expect(wrapper).toMatchSnapshot();
    });

    it('calls onClickConnect handler', () => {
        const data = {
            ...info,
            callsValue: 0,
            trafficValue: 0,
        };
        const fn = jest.fn();
        const wrapper = shallow(<ProductTile {...tariff} onClickConnect={fn} />);

        wrapper.find(Buy).dive().find(TextLink).simulate('click', {});
        expect(fn).toBeCalledWith(data, {});
    });

    it('calls onClickConnect handler with servicePacks', () => {
        const data = {
            ...info,
            callsValue: 300,
            trafficValue: 6,
        };
        const fn = jest.fn();
        const wrapper = shallow(<ProductTile {...tariff} onClickConnect={fn} servicePacks={servicePacks} />);

        wrapper.find(Buy).dive().find(TextLink).simulate('click', {});
        expect(fn).toBeCalledWith(data, {});
    });

    it('calls handleClickBuy handler', () => {
        const data = {
            ...info,
            callsValue: 0,
            trafficValue: 0,
            shopTag: 'choose',
            price: '100',
            unitValue: '₽',
            unitExtra: 'за 30 дней',
        };
        const fn = jest.fn();
        const wrapper = shallow(<ProductTile {...tariff} onClickBuy={fn} />);

        wrapper.find(Buy).dive().find(Button).simulate('click', {});
        expect(fn).toBeCalledWith(data, {});
    });

    it('calls handleClickBuy handler with servicePacks', () => {
        const data = {
            ...info,
            callsValue: 300,
            trafficValue: 6,
            shopTag: 'choose_11',
            price: '100',
            unitValue: '₽',
            unitExtra: 'за 30 дней',
        };
        const fn = jest.fn();
        const wrapper = shallow(<ProductTile {...tariff} onClickBuy={fn} servicePacks={servicePacks} />);

        wrapper.find(Buy).dive().find(Button).simulate('click', {});
        expect(fn).toBeCalledWith(data, {});
    });

    it('calls onClickMore handler', () => {
        const fn = jest.fn();
        const wrapper = shallow(<ProductTile {...tariff} onClickMore={fn} />);

        wrapper.find('.mfui-product-tile__title-link').dive().first().simulate('click', {});
        expect(fn).toBeCalledWith(info, {});

        wrapper.find('.mfui-product-tile__detail-link').dive().first().simulate('click', {});
        expect(fn).toBeCalledWith(info, {});
    });

    it('calls onClickBubble handler', () => {
        const fn = jest.fn();
        const wrapper = mount(<ProductTile {...tariff} onClickBubble={fn} />);

        wrapper.find('.mfui-product-tile-options__bubble-trigger').simulate('click');
        expect(fn).toBeCalledWith(info);
    });

    it('calls handleChangeCalls handler', () => {
        const wrapper = mount(<ProductTile {...tariff} servicePacks={servicePacks} />);
        const instance = wrapper.instance() as ProductTile;

        instance.handleChangeCalls({} as React.ChangeEvent<HTMLInputElement>, '300');
        wrapper.find('.mfui-product-switcher__item').first().simulate('click');

        compareRestState(wrapper);
        expect(wrapper.state('callsValue')).toEqual(300);
    });

    it('calls handleChangeTraffic handler', () => {
        const wrapper = mount(<ProductTile {...tariff} servicePacks={servicePacks} />);
        const instance = wrapper.instance() as ProductTile;

        instance.handleChangeTraffic({} as React.ChangeEvent<HTMLInputElement>, '6');
        wrapper.find('.mfui-product-switcher__item').last().simulate('click');

        compareRestState(wrapper);
        expect(wrapper.state('trafficValue')).toEqual(6);
    });

    it('should return object from getRestState call', () => {
        const [ currentPack ] = servicePacks;
        const wrapper = shallow(<ProductTile {...tariff} servicePacks={servicePacks} />);
        const instance = wrapper.instance() as ProductTile;
        const receivedObject = instance.getRestState(currentPack);
        const expectedObject = {
            currentPack: currentPack,
            price: currentPack.payment.value,
            discount: currentPack.payment.discount,
            options: currentPack.options,
            buyLink: currentPack.buyLink
        };

        expect(receivedObject).toEqual(expectedObject);
    });


    it('should return object from getCurrentPack call', () => {
        const [ currentPack ] = servicePacks;
        const wrapper = shallow(<ProductTile {...tariff} servicePacks={servicePacks} />);
        const instance = wrapper.instance() as ProductTile;
        const receivedObject = instance.getCurrentPack(300, 6);

        expect(receivedObject).toEqual(currentPack);
    });

    it('should return object from getSwitcherValues call', () => {
        const wrapper = shallow(<ProductTile {...tariff} servicePacks={servicePacks} />);
        const instance = wrapper.instance() as ProductTile;
        const receivedObject = instance.getSwitcherValues();
        const expectedObject = {
            calls: ['300'],
            traffic: ['6'],
        };

        expect(receivedObject).toEqual(expectedObject);
    });

    it('should return link from formHashLink call', () => {
        const wrapper = shallow(<ProductTile {...tariff} servicePacks={servicePacks} />);
        const instance = wrapper.instance() as ProductTile;
        const link = instance.formHashLink('/tariffs/all/vybiray.html', '');

        expect(link).toEqual('/tariffs/all/vybiray.html');
    });

    it('should return hashLink from formHashLink call', () => {
        const wrapper = shallow(<ProductTile {...tariff} servicePacks={servicePacks} />);
        const instance = wrapper.instance() as ProductTile;
        const link = instance.formHashLink('/tariffs/all/vybiray.html', 'choose');

        expect(link).toEqual('/tariffs/all/vybiray.html#choose');
    });
});
