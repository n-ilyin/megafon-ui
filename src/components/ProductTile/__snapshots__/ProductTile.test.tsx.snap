// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTile /> renders ProductTile 1`] = `
<div
  className="mfui-product-tile"
>
  <div
    className="mfui-product-tile__content"
  >
    <h2
      className="mfui-product-tile__title"
    >
      <TextLink
        className="mfui-product-tile__title-link"
        color="black"
        href="/tariffs/all/vybiray.html"
        onClick={[Function]}
        target="_blank"
        underlineStyle="none"
        underlineVisibility="hover"
      >
        Включайся! Выбирай
      </TextLink>
    </h2>
    <div
      className="mfui-product-tile__link-wrapper"
    >
      <TextLink
        className="mfui-product-tile__detail-link"
        color="blue"
        href="/tariffs/all/vybiray.html"
        onClick={[Function]}
        target="_blank"
        underlineStyle="solid"
        underlineVisibility="hover"
      >
        Подробнее
      </TextLink>
    </div>
    <ProductTileCashback
      title="Кэшбэк"
      unit="%"
      value={20}
    />
    <ProductTilePrice
      discount="100"
      title="При покупке новой SIM–карты"
      unitExtra="за 30 дней"
      unitValue="₽"
      value="3000"
    />
    <div
      className="mfui-product-tile__constructor"
    >
      <ProductTileStatic
        packs={
          Array [
            Object {
              "isDelim": false,
              "title": "Звонки на все номера России",
              "unit": "ГБ",
              "value": 10,
            },
            Object {
              "isDelim": true,
              "title": "Безлимитный интернет",
              "unit": "",
              "value": 0,
            },
          ]
        }
      />
      <ProductTileOptions
        options={
          Array [
            Object {
              "caption": "Пакеты кино и сериалов",
              "svgIcon": <test-file-stub />,
              "title": "МегаФон ТВ",
            },
          ]
        }
      />
    </div>
    <ProductTileOptions
      className="mfui-product-tile__second-params"
      onClickBubble={[Function]}
      options={
        Array [
          Object {
            "footnote": "Участвуйте в программе",
            "svgIcon": <test-file-stub />,
            "title": "Кэшбэк 20%",
            "unit": "",
            "value": "",
          },
          Object {
            "footnote": "",
            "svgIcon": <test-file-stub />,
            "title": "Безлимитные SMS",
            "unit": "₽ за 30 дней",
            "value": "100",
          },
        ]
      }
    />
  </div>
  <ProductTileBuy
    buyButton={
      Object {
        "buttonPassiveColor": "green",
        "buyButtonText": "Купить",
        "buyLink": "/zakaz/?tariff=choose",
        "showBuyButton": true,
      }
    }
    connectButton={
      Object {
        "connectButtonText": "Перейти на тариф",
        "connectLink": "/tariffs/all/vybiray.html",
        "showConnectButton": true,
      }
    }
    onClickBuy={[Function]}
    onClickConnect={[Function]}
  />
</div>
`;

exports[`<ProductTile /> renders ProductTile 2`] = `
<div
  className="mfui-product-tile mfui-product-tile_constructor"
>
  <div
    className="mfui-product-tile__content"
  >
    <h2
      className="mfui-product-tile__title"
    >
      <TextLink
        className="mfui-product-tile__title-link"
        color="black"
        href="/tariffs/all/vybiray.html"
        onClick={[Function]}
        target="_blank"
        underlineStyle="none"
        underlineVisibility="hover"
      >
        Включайся! Выбирай
      </TextLink>
    </h2>
    <div
      className="mfui-product-tile__link-wrapper"
    >
      <TextLink
        className="mfui-product-tile__detail-link"
        color="blue"
        href="/tariffs/all/vybiray.html#choose_11"
        onClick={[Function]}
        target="_blank"
        underlineStyle="solid"
        underlineVisibility="hover"
      >
        Подробнее
      </TextLink>
    </div>
    <ProductTileCashback
      title="Кэшбэк"
      unit="%"
      value={20}
    />
    <ProductTilePrice
      discount="100"
      title="При покупке новой SIM–карты"
      unitExtra="за 30 дней"
      unitValue="₽"
      value="450"
    />
    <div
      className="mfui-product-tile__constructor"
    >
      <ProductTileDynamic
        currentPack={
          Object {
            "buyLink": "/zakaz/?tariff=choose_11",
            "calls": Object {
              "unit": "минут",
              "value": 300,
            },
            "options": Array [
              Object {
                "footnote": "Участвуйте в программе",
                "svgIcon": <test-file-stub />,
                "title": "Кэшбэк 20%",
                "unit": "",
                "value": "",
              },
              Object {
                "footnote": "Подключите в разделе «Услуги и опции»",
                "svgIcon": <test-file-stub />,
                "title": "Безлимитные SMS",
                "unit": "₽ за 30 дней",
                "value": "55",
              },
            ],
            "payment": Object {
              "discount": "",
              "value": "450",
            },
            "shopTag": "choose_11",
            "traffic": Object {
              "unit": "ГБ",
              "value": 6,
            },
          }
        }
        onChangeCalls={[Function]}
        onChangeTraffic={[Function]}
        startCallsIndex={0}
        startTrafficIndex={0}
        switcher={
          Object {
            "calls": Array [
              "300",
            ],
            "traffic": Array [
              "6",
            ],
          }
        }
      />
      <ProductTileOptions
        options={
          Array [
            Object {
              "caption": "Пакеты кино и сериалов",
              "svgIcon": <test-file-stub />,
              "title": "МегаФон ТВ",
            },
          ]
        }
      />
    </div>
    <ProductTileOptions
      className="mfui-product-tile__second-params"
      onClickBubble={[Function]}
      options={
        Array [
          Object {
            "footnote": "Участвуйте в программе",
            "svgIcon": <test-file-stub />,
            "title": "Кэшбэк 20%",
            "unit": "",
            "value": "",
          },
          Object {
            "footnote": "Подключите в разделе «Услуги и опции»",
            "svgIcon": <test-file-stub />,
            "title": "Безлимитные SMS",
            "unit": "₽ за 30 дней",
            "value": "55",
          },
        ]
      }
    />
  </div>
  <ProductTileBuy
    buyButton={
      Object {
        "buttonPassiveColor": "green",
        "buyButtonText": "Купить",
        "buyLink": "/zakaz/?tariff=choose_11",
        "showBuyButton": true,
      }
    }
    connectButton={
      Object {
        "connectButtonText": "Перейти на тариф",
        "connectLink": "/tariffs/all/vybiray.html",
        "showConnectButton": true,
      }
    }
    onClickBuy={[Function]}
    onClickConnect={[Function]}
  />
</div>
`;

exports[`<ProductTile /> renders ProductTile with topBadgeTitle 1`] = `
<div
  className="mfui-product-tile mfui-product-tile_constructor"
>
  <ProductTileHint
    title="Тариф-конструктор от 450 ₽"
  />
  <div
    className="mfui-product-tile__content"
  >
    <h2
      className="mfui-product-tile__title"
    >
      <TextLink
        className="mfui-product-tile__title-link"
        color="black"
        href="/tariffs/all/vybiray.html"
        onClick={[Function]}
        target="_blank"
        underlineStyle="none"
        underlineVisibility="hover"
      >
        Включайся! Выбирай
      </TextLink>
    </h2>
    <div
      className="mfui-product-tile__link-wrapper"
    >
      <TextLink
        className="mfui-product-tile__detail-link"
        color="blue"
        href="/tariffs/all/vybiray.html#choose_11"
        onClick={[Function]}
        target="_blank"
        underlineStyle="solid"
        underlineVisibility="hover"
      >
        Подробнее
      </TextLink>
    </div>
    <ProductTileCashback
      title="Кэшбэк"
      unit="%"
      value={20}
    />
    <ProductTilePrice
      discount="100"
      title="При покупке новой SIM–карты"
      unitExtra="за 30 дней"
      unitValue="₽"
      value="450"
    />
    <div
      className="mfui-product-tile__constructor"
    >
      <ProductTileDynamic
        currentPack={
          Object {
            "buyLink": "/zakaz/?tariff=choose_11",
            "calls": Object {
              "unit": "минут",
              "value": 300,
            },
            "options": Array [
              Object {
                "footnote": "Участвуйте в программе",
                "svgIcon": <test-file-stub />,
                "title": "Кэшбэк 20%",
                "unit": "",
                "value": "",
              },
              Object {
                "footnote": "Подключите в разделе «Услуги и опции»",
                "svgIcon": <test-file-stub />,
                "title": "Безлимитные SMS",
                "unit": "₽ за 30 дней",
                "value": "55",
              },
            ],
            "payment": Object {
              "discount": "",
              "value": "450",
            },
            "shopTag": "choose_11",
            "traffic": Object {
              "unit": "ГБ",
              "value": 6,
            },
          }
        }
        onChangeCalls={[Function]}
        onChangeTraffic={[Function]}
        startCallsIndex={0}
        startTrafficIndex={0}
        switcher={
          Object {
            "calls": Array [
              "300",
            ],
            "traffic": Array [
              "6",
            ],
          }
        }
      />
      <ProductTileOptions
        options={
          Array [
            Object {
              "caption": "Пакеты кино и сериалов",
              "svgIcon": <test-file-stub />,
              "title": "МегаФон ТВ",
            },
          ]
        }
      />
    </div>
    <ProductTileOptions
      className="mfui-product-tile__second-params"
      onClickBubble={[Function]}
      options={
        Array [
          Object {
            "footnote": "Участвуйте в программе",
            "svgIcon": <test-file-stub />,
            "title": "Кэшбэк 20%",
            "unit": "",
            "value": "",
          },
          Object {
            "footnote": "Подключите в разделе «Услуги и опции»",
            "svgIcon": <test-file-stub />,
            "title": "Безлимитные SMS",
            "unit": "₽ за 30 дней",
            "value": "55",
          },
        ]
      }
    />
  </div>
  <ProductTileBuy
    buyButton={
      Object {
        "buttonPassiveColor": "green",
        "buyButtonText": "Купить",
        "buyLink": "/zakaz/?tariff=choose_11",
        "showBuyButton": true,
      }
    }
    connectButton={
      Object {
        "connectButtonText": "Перейти на тариф",
        "connectLink": "/tariffs/all/vybiray.html",
        "showConnectButton": true,
      }
    }
    onClickBuy={[Function]}
    onClickConnect={[Function]}
  />
</div>
`;
